### ISSUE TITLE
Add Multizim search to android

### ISSUE THREAD

**Issue by mhutti1** [kiwix/kiwix-xulrunner#364]
We should implement multizim search as in the IOS app into the android app.

The ZIM files that are searchable should be a user selected subset of the available ZIM files.

Memory implications might need to be looked into if we use the JNI to open multiple files at once. I am not too sure at the moment.

**Comment by mhutti1** [comment]
I have actually got much further much faster than I thought I could. Although still a prototype I have successfully implemented multizim search with only a few issues.

https://github.com/kiwix/kiwix/tree/android_multizim_search

At the moment I can see 3 potential issues.

First, memory:

- [ ] Work out memory implications of storing multiple zim readers

The additional searching has made searches take considerably longer as such:

- [ ] Searches should be added to the list as returned

Files with duplicate titles currently default to the physically open zim file. 

- [ ] Both files should be displayed along with their ZIM file and the user should be given the option of which to choose

In addition to this currently to load the multizims a user must visit the content management screen. We should remove this requirement (and implement functions to allow the adding of additional readers to the array without swapping the current reader out)

PS: I am a complete C++ novice as such my code may be dreadful, feel free to point out any flaws.
